pkg/authorizer/token_retriever_wi.go

```diff
@@ -0,0 +1,45 @@
+package authorizer
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/confidential"
+)
+
+const authorityHost = "https://login.microsoftonline.com/"
+
+// Get auth token from service account token
+func AcquireARMTokenFromServiceAccountToken(ctx context.Context, tenantID, clientID string) (base.authResult, error) {
+
+	cred := confidential.NewCredFromAssertionCallback(func(context.Context, confidential.AssertionRequestOptions) (string, error) {
+		return readJWTFromFS()
+	})
+
+	confidentialClientApp, err := confidential.New(
+		clientID,
+		cred,
+		confidential.WithAuthority(fmt.Sprintf("%s%s/oauth2/token", authorityHost, tenantID)))
+	if err != nil {
+		return "", fmt.Errorf("Unable to get new confidential client app: %w", err)
+	}
+
+	authResult, err := confidentialClientApp.AcquireTokenByCredential(ctx, []string{"/.default"})
```

Comment:
You must include resource. See lines 35-42 [here](https://github.com/niontive/wi-acrpull/blob/b41f035bfd442a94706013410510a1b0405ae5ff/pkg/authorizer/token.go#L35).

Resource `"https://management.azure.com/"` is ARM; this is because we get an ARM token.